### PR TITLE
Incorporate Web REST API HTTP response codes

### PR DESCRIPTION
**Pull request by lamaks**
#### Description of changes

Incorporate Web REST API HTTP response codes.
Update scan run REST API client response.
Delete unused code.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/azure-services/src/register-azure-services-to-container.ts [hunk 1]
```diff
@@ -79,9 +78,16 @@ export function registerAzureServicesToContainer(container: Container, credentia
...
+        let accountName: string;
+        let accountKey: string;
+        if (process.env.AZURE_STORAGE_NAME !== undefined && process.env.AZURE_STORAGE_KEY !== undefined) {
+            accountName = process.env.AZURE_STORAGE_NAME;
+            accountKey = process.env.AZURE_STORAGE_KEY;
+        } else {
+            const secretProvider = context.container.get(SecretProvider);
+            accountName = await secretProvider.getSecret(secretNames.storageAccountName);
+            accountKey = await secretProvider.getSecret(secretNames.storageAccountKey);
+        }
```
Comment (manishsat) [inline packages/azure-services/src/register-azure-services-to-container.ts]: This block seems to be glue code to me, setupSingletonQueueServiceURLProvider() method is also doing the same thing. Can we move this to a common place?

File: packages/web-api/src/controllers/batch-scan-result-controller.ts [hunk 2]
```diff
@@ -26,9 +27,13 @@ export class BatchScanResultController extends BaseScanResultController {
 
     public async handleRequest(): Promise<void> {
         const payload = this.tryGetPayload<ScanBatchRequest[]>();
+        if (payload === undefined) {
+            return;
```
Comment (manishsat) [inline packages/web-api/src/controllers/batch-scan-result-controller.ts]: Is this going to return 200?

File: packages/web-api/src/controllers/batch-scan-result-controller.ts [hunk 3]
```diff
@@ -26,9 +27,13 @@ export class BatchScanResultController extends BaseScanResultController {
...
+        }
+
         if (isEmpty(payload)) {
             this.context.res = {
-                status: 422, // Unprocessable Entity,
+                status: 204, // No Content
```
Comment (manishsat) [inline packages/web-api/src/controllers/batch-scan-result-controller.ts]: This if and the one just above it, can be moved to WebController? I am assuming we need to do this for all type of APIController.

File: packages/web-api/src/controllers/batch-scan-result-controller.ts (hunk 2 above)
Comment (lamaks) [reply packages/web-api/src/controllers/batch-scan-result-controller.ts]: Will return 400 InvalidJsonDocument

File: packages/web-api/src/api-contracts/scan-result-response.ts [hunk 4]
```diff
@@ -20,6 +17,12 @@ export type ScanResultResponse =
       }
     | ScanResultErrorResponse;
 
+export interface ScanResultErrorResponse {
+    scanId: string;
+    code: WebApiErrorCodeName;
+    error: string;
+}
```
Comment (devaradhanm) [inline packages/web-api/src/api-contracts/scan-result-response.ts]: we may need to send {
url,
code, 
error
}
& not scan id if there is error

File: packages/web-api/src/controllers/batch-scan-result-controller.ts (hunk 3 above)
Comment (lamaks) [reply packages/web-api/src/controllers/batch-scan-result-controller.ts]: Handled as part of ApiController

File: packages/web-api/src/api-contracts/scan-result-response.ts (hunk 4 above)
Comment (lamaks) [reply packages/web-api/src/api-contracts/scan-result-response.ts]: The ScanRunResponse is type when we have error before we assign scan id. Other cases will have scan id already so we can use above type.

**Pull request merged**